client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -165,6 +166,18 @@ static Request createIndex(CreateIndexRequest createIndexRequest) throws IOExcep
         return new Request(HttpPut.METHOD_NAME, endpoint, parameters.getParams(), entity);
     }
 
+    static Request putMapping(PutMappingRequest putMappingRequest) throws IOException {
+        String endpoint = endpoint(putMappingRequest.indices(), "_mapping", putMappingRequest.type());
```

Comment:
Do we want to support `putMappingRequest.getConcreteIndex()` here? If so, how/when should it be used? Does it make sense to support in client-specific code?

Reply:
no that's an internal concept leaked to our external API. Maybe check in `RestHighLevelClient` that `concreteIndex` is not set, as it will be ignored.